### PR TITLE
feat: empty: adds empty state component

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:
Adds Empty state  component

https://user-images.githubusercontent.com/99700808/173470221-d8c73d33-cdea-41fa-ba0e-132f4e19f202.mp4

## JIRA TASK (Eightfold Employees Only):
ENG-13961

## CHANGE TYPE:

-   [ ] Bugfix Pull Request
-   [X] Feature Pull Request

## TEST COVERAGE:

-   [ ] Tests for this change already exist
-   [X] I have added unittests for this change

## TEST PLAN:
pull pr changes locally, run and go to http://localhost:2022/?path=/story/empty--no-data

### CODE REVIEW

File: src/components/Empty/Empty.stories.tsx
Comment (jhoward-eightfold) [inline src/components/Empty/Empty.stories.tsx:83]: if we allow implementations to specify colors in this way, it may make for more difficult theming. Might it be possibly to specify colors as css var names?

File: src/components/Empty/Empty.test.tsx
Comment (jhoward-eightfold) [inline src/components/Empty/Empty.test.tsx:15]: we need to begin writing more meaningful tests to avoid growing technical debt

File: src/components/Empty/Empty.types.ts
Comment (jhoward-eightfold) [inline src/components/Empty/Empty.types.ts:4]: do we maintain a standard approach of capitalizing enum keys?

File: src/components/Empty/SVG/DefaultEmptyDataImg.tsx
Comment (jhoward-eightfold) [inline src/components/Empty/SVG/DefaultEmptyDataImg.tsx:22]: we are going to need a mechanism by which we apply svg fills via class names so we can support theming of svgs.  It might be the case that we want these empty state graphics to change colors when switching to dark mode.

File: src/components/Empty/rtl.scss
Comment (jhoward-eightfold) [inline src/components/Empty/rtl.scss:2]: interesting. looking forward to seeing this in action

**Review by jhoward-eightfold — APPROVED** [review]

File: src/components/Empty/Empty.stories.tsx
Comment (dkilgore-eightfold) [reply src/components/Empty/Empty.stories.tsx:83]: We can accomplish this via props, the component also supports img src urls. So this will require more specification. Will add a story to the backlog.

File: src/components/Empty/Empty.test.tsx
Comment (dkilgore-eightfold) [reply src/components/Empty/Empty.test.tsx:15]: Agreed, stubbing this one in to unblock Table. There are really only 2 this component needs, may add them before merging.

File: src/components/Empty/Empty.types.ts
Comment (dkilgore-eightfold) [reply src/components/Empty/Empty.types.ts:4]: No, the other contributor's favor lowercase, but the components I initially added use the .NET pattern of sentence casing. Those should probably be updated to map to the majority.

Comment (jhoward-eightfold) [reply src/components/Empty/Empty.types.ts:4]: Ack

File: src/components/Empty/SVG/DefaultEmptyDataImg.tsx
Comment (dkilgore-eightfold) [reply src/components/Empty/SVG/DefaultEmptyDataImg.tsx:22]: This can be done via props passed to the SVG component. Can also be done in SASS like the other components via the theme selectors

File: src/components/Empty/Empty.test.tsx
Comment (jhoward-eightfold) [reply src/components/Empty/Empty.test.tsx:15]: Ack

File: src/components/Empty/rtl.scss
Comment (dkilgore-eightfold) [reply src/components/Empty/rtl.scss:2]: Ya, we'll need to add the rtl SASS file to each component, start ad-hoc testing to get ahead of future support of AR, HE.

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**
